### ISSUE TITLE
Integrate Armstrong Validation into the spec PR check suite.

### ISSUE THREAD

**Issue by konrad-jamrozik** [Azure/azure-rest-api-specs#28773]

**Comment by maririos** [comment]
I am curious on understanding this process and the implication it will have on our users. There was a conversation on email where it was communicated that the check was going to be optional so not everyone will require it. What is the criteria for the users?

**Comment by ms-zhenhua** [comment]
> I am curious on understanding this process and the implication it will have on our users. There was a conversation on email where it was communicated that the check was going to be optional so not everyone will require it. What is the criteria for the users?

There are 2 types of Armstrong validations for users:
1. When do the users need to use Armstrong to do API Test? Currently, we only require new APIs against the main branch of the public swagger repo needs to do API Test, but the scope may change in future.
2. The submitted API Test configurations must not have any explicit credential values. This check should be required for all the swagger repo.

**Comment by maririos** [comment]
Thank you @ms-zhenhua 

> When do the users need to use Armstrong to do API Test? Currently, we only require new APIs against the main branch of the public swagger repo needs to do API Test, but the scope may change in future.

How will the user know that? Besides a failure in the CI, how areyou thinking on letting the user know that this is a required step now?

**Comment by ms-zhenhua** [comment]
> Thank you @ms-zhenhua
> 
> > When do the users need to use Armstrong to do API Test? Currently, we only require new APIs against the main branch of the public swagger repo needs to do API Test, but the scope may change in future.
> 
> How will the user know that? Besides a failure in the CI, how are you thinking on letting the user know that this is a required step now?

RP API Review Team is drafting the API Test workflow and will schedule a meeting with RP teams. After reaching an agreement, API Test will be included as a part of RP API review and the RP team can submit the API Test configurations for their new APIs. Before that, I need to make Armstrong Validation as a required check so that it can help identify the security risks when RP teams start to submit their Terraform configurations.

**Comment by mikeharder** [comment]
> BTW, is it possible to create a pipeline to test this feature on my dev branch as what openapi-alps supports? I need to test what's the final result displayed in GitHub.

Yes, you'd want to do the following:

1. Move the changes in your PR from a branch in your fork "ms-zhenhua", to a branch in the main repo "azure-rest-api-specs" (named something like "ms-zhenhua/armstrong-validation").  Typically we recommend using a fork, but for new checks, it's easier to test from a branch in the main repo.
2. Create a new PR from the branch, and close #28803.
3. Azure SDK creates a pipeline in our "playground" project that points to your branch, and runs your pipeline only on PRs to your branch (not to main).
4. You can test the pipeline by creating draft PRs to your branch (rather than to main).
5. When you think the check is ready for prod, merge your PR to azure-rest-api-specs/main, and we will move the DevOps pipeline to prod (and make it a required check).

**Comment by mikeharder** [comment]
@ms-zhenhua: I just left comments on 28803, feel free to reply in that PR, or if you open a new PR I can move my comments.

**Comment by ms-zhenhua** [comment]
> @ms-zhenhua: I just left comments on 28803, feel free to reply in that PR, or if you open a new PR I can move my comments. Once you have opened the new PR we can create the test pipeline.

@mikeharder, thank you for the review. I created a new PR(https://github.com/Azure/azure-rest-api-specs/pull/28829) to replace with the original one and resolved the comments. Could you please guide me how to do the step 3 of `Azure SDK creates a pipeline in our "playground" project that points to your branch, and runs your pipeline only on PRs to your branch (not to main)` so that I can do some test?

Besides that, I have another problem about the first point of [this comment](https://github.com/Azure/azure-rest-api-specs/issues/28773#issuecomment-2071680565) to confirm with you. At first, I decide to implement the logic with the following steps:
1. let the RP API reviewer or pipeline task to add a label `Armstrong-Validation` to mark whether a PR needs to do API Test and blocks this PR.
2. The RP team complete the API Test and submit the test result.(The result contains 2 parts: one are the test configurations which will be submitted with the PR, the other part are test results which will be submitted in the PR comments)
3. The RP API reviewer reviews the test result and add the "Approved-Armstrong-Validation" label to unblock the PR.

Since the label method is obsolete, is there other mechanism which can support such manual controlled validation?

**Comment by maririos** [comment]
FYI @ladonnaq and @josefree
